lib/plug/crypto/message_encryptor.ex

```diff
@@ -97,10 +285,93 @@ defmodule Plug.Crypto.MessageEncryptor do
     end
   end
 
-  defp trim_secret(secret) do
-    case byte_size(secret) do
-      large when large > 32 -> :binary.part(secret, 0, 32)
-      _ -> secret
+  ## Helpers
+
+  defp encode_token(protected, encrypted_key, iv, cipher_text, cipher_tag) do
+    protected
+    |> Base.url_encode64(padding: false)
+    |> Kernel.<>(".")
+    |> Kernel.<>(Base.url_encode64(encrypted_key, padding: false))
+    |> Kernel.<>(".")
+    |> Kernel.<>(Base.url_encode64(iv, padding: false))
+    |> Kernel.<>(".")
+    |> Kernel.<>(Base.url_encode64(cipher_text, padding: false))
+    |> Kernel.<>(".")
+    |> Kernel.<>(Base.url_encode64(cipher_tag, padding: false))
+  end
+
+  defp decode_token(token) do
+    case String.split(token, ".", parts: 5) do
+      [protected, encrypted_key, iv, cipher_text, cipher_tag] ->
+        with {:ok, protected}     <- Base.url_decode64(protected, padding: false),
```

Comment:
Can we leave a TODO here so we use `with/else` once we depend on Elixir v1.3 only?